### ISSUE TITLE
Artists with one or more Pseudonym's with same mb-artist-id aren't correct handeld during catalog update

### ISSUE THREAD

**Issue by SePHTaN** [ampache/ampache#2806]
Why i am scared:
I just ran an update to my database cause i exchanged some of my albums with reencoded versions with higher bitrate.
Now i have several artists put together ("Die Toten Hosen" and "Die Roten Rosen" for example) that were correct separated before. All Albums of "The Toten Hosen" are now from "The Rote Rosen" except the one album i updated.
Also "The Toten Hosen" are missing musicbrainz-ids in the database i suspect having to do with an Feature Request just some weeks ago.
Update from Tags of the existing albums of The Toten Hosen (the Tags are totally correct just checked it) only say "Es wird keine Aktualisierung benötigt./ No actualisation needed" .

_Originally posted by @SePHTaN in https://github.com/ampache/ampache/issues/2799#issuecomment-791805151_

**Comment by lachlan-00** [comment]
It's actually always joined them; the Artist check function will search by mbid and return matches.

```
        if ($mbid !== '') {
            $sql        = 'SELECT `id` FROM `artist` WHERE `mbid` = ?';
            $db_results = Dba::read($sql, array($mbid));

            if ($row = Dba::fetch_assoc($db_results)) {
                $artist_id = (int)$row['id'];
                $exists    = true;
            }
        }

```

**Comment by SePHTaN** [comment]
Update on this : 
I yesterday cleaned my catalog from all albums from both bands "Die Toten Hosen" and "Die Rote Rosen" by deleting all files from my storage and then ran 'clean_all_catalog'.
I assume i missed some sampler containing at least one song from "Die Roten Rosen" and didn't checked the database to be sure everything from them disappeared.
Last hour i added in all albums from "The Toten Hosen" and that's the result:
This album is totally correct added solely mbid for artist is missing
![totehosen](https://user-images.githubusercontent.com/10261043/110397914-06e1fc80-8073-11eb-9c7c-601c65be2564.jpg)
Those should all be from "The Toten Hosen" because i didn't yet added the albums from "Die Roten Rosen":
![roterosen](https://user-images.githubusercontent.com/10261043/110397925-09dced00-8073-11eb-88a7-e313b1b9ebf6.jpg)

But as you can see ampache in fact is able to add them both, not totally correct one is missing the mbid.
So as you say ampache won't add mbid duplicates, hence it just leaves it empty.
Allowing mbid duplicates should be an option, it has to be checked that the artist name is different than, at least as artist like that exist in reality.

**Comment by lachlan-00** [comment]
When you don't have an mbid it will look up by name which might not always match the artist which has the mbid. 
```
$sql        = 'SELECT `id`, `mbid` FROM `artist` WHERE `name` LIKE ?';
```

The server will include all of those with the same mbid and will use the name discovered first. (Die Roten Rosen for you)
What this means is every Die Toten Hosen album with no mbid is not considered the same.

What i would do is manually rename Die Roten Rosen (Edit Artist) and it update the files missing mbid to the same artist using verify/update from tags.

What Ampache should do is in these situations is pick the name by majority or by querying musicbrainz for the release group name and rename accordingly. It would be good to write a function to update this detail from musicbrainz for the artists similar to the last.fm update.

That answer was a bit long but the short answer is that the MBID is something associated with a single name, with aliases that i don't think should be separated into different groups.

Musicbrainz and ampache both treat aliases as a single artist. Even though they are different names they are the same object.
![image](https://user-images.githubusercontent.com/1305249/110448642-19a90f80-810d-11eb-8f4f-fd299226dc5c.png)

**Comment by SePHTaN** [comment]
You are totally right with musicbrainz treating them as the same artist at least regarding MBID but they do respect the fact that some albums are from the pseudonym and list them, as you discovered yourself, under correct artist name even if MBID is the same.
>What this means is every Die Toten Hosen album with no mbid is not considered the same.

I generally tag my albums with Musicbrainz Picard, so all my albums have an MBID.
Those rare cases, about 4 at the moment, where the album isn't listed with musicbrainz sit on my lokal drive waiting they get added by someone else to musicbrainz. 

>What i would do is manually rename Die Roten Rosen (Edit Artist)
I already did this yesterday directly in the database with phpmyadmin with no result.
But i have to admit i did not run verify/update from tags, i will try this later when i picked up my daugher from soccer training.

>Musicbrainz and ampache both treat aliases as a single artist. Even though they are different names they are the same object.

I would like if ampache would be able to take in account that there is a pseudonym it's totally ok and logical absolutly right if they are stored as one artist with one MBID.
To achieve this there has to be second name (the pseudonym) stored with the artist with that MBID, and a way to store with albums/songs not only the id for the artist with the respective MBID but too if it uses the firstname or the second name.
That's what i would like ampache to be able to, the simple reason is i do have one or two more artists with the same constellation.
One MBID two names, though they have only few albums compared to  "Die Toten Hosen".
